### My post 

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
weather_df = pd.read_csv('data/philadelphia_weather_2005_to_2025.csv')
weather_df = weather_df.assign(date=pd.to_datetime(weather_df['date']))

#### Inspect the data

In [3]:
weather_df.columns

Index(['day', 'year', 'high', 'low', 'rain', 'snow', 'date', 'day_num',
       'month_num', 'dow', 'dow_name', 'month'],
      dtype='str')

In [4]:
weather_df.head()

day  year  high  low  rain  snow       date  day_num  month_num  dow  \
0  January 1  2005    64   39  0.00   0.0 2005-01-01        1          1    5   
1  January 2  2005    47   33  0.01   0.0 2005-01-02        2          1    6   
2  January 3  2005    57   40  0.06   0.0 2005-01-03        3          1    0   
3  January 4  2005    55   44  0.00   0.0 2005-01-04        4          1    1   
4  January 5  2005    47   36  0.62   0.0 2005-01-05        5          1    2   

    dow_name    month  
0   Saturday  January  
1     Sunday  January  
2     Monday  January  
3    Tuesday  January  
4  Wednesday  January

In [5]:
weather_df['year'].value_counts()

year
2008    366
2020    366
2024    366
2016    366
2012    366
2009    365
2007    365
2006    365
2005    365
2013    365
2014    365
2011    365
2010    365
2017    365
2015    365
2019    365
2018    365
2021    365
2022    365
2023    365
2025    365
Name: count, dtype: int64

## 1) What was the weather like in Philadelphia the day of my birthday (01-30) for each year since 2006-2025?
  ## * a. What was the average fahrenheit for the day?
  ## * b. Did it rain?
  ## * c. Did it snow?
  ## * d. What day of the week was it? 

In [8]:
weather_df.query('day =="January 30"')

day  year  high  low  rain  snow       date  day_num  month_num  \
29    January 30  2005    36   26  0.15   1.3 2005-01-30       30          1   
394   January 30  2006    62   37  0.00   0.0 2006-01-30       30          1   
759   January 30  2007    39   20  0.02   0.2 2007-01-30       30          1   
1124  January 30  2008    49   29  0.05   0.0 2008-01-30       30          1   
1490  January 30  2009    36   22  0.00   0.0 2009-01-30       30          1   
1855  January 30  2010    21   16  0.10   2.1 2010-01-30       30          1   
2220  January 30  2011    39   18  0.00   0.0 2011-01-30       30          1   
2585  January 30  2012    42   33  0.00   0.0 2012-01-30       30          1   
2951  January 30  2013    68   38  0.32   0.0 2013-01-30       30          1   
3316  January 30  2014    28   10  0.00   0.0 2014-01-30       30          1   
3681  January 30  2015    42   20  0.00   0.1 2015-01-30       30          1   
4046  January 30  2016    41   26  0.00   0.0 2016-01-30       30          1   
4412  January 30  2017    39   27  0.02   0.8 2017-01-30       30          1   
4777  January 30  2018    38   22  0.14   0.3 2018-01-30       30          1   
5142  January 30  2019    35    7  0.01   0.2 2019-01-30       30          1   
5507  January 30  2020    39   25  0.00   0.0 2020-01-30       30          1   
5873  January 30  2021    35   19  0.00   0.0 2021-01-30       30          1   
6238  January 30  2022    27   11  0.00   0.0 2022-01-30       30          1   
6603  January 30  2023    54   37  0.00   0.0 2023-01-30       30          1   
6968  January 30  2024    39   34  0.00   0.0 2024-01-30       30          1   
7334  January 30  2025    42   29  0.00   0.0 2025-01-30       30          1   

      dow   dow_name    month  
29      6     Sunday  January  
394     0     Monday  January  
759     1    Tuesday  January  
1124    2  Wednesday  January  
1490    4     Friday  January  
1855    5   Saturday  January  
2220    6     Sunday  January  
2585    0     Monday  January  
2951    2  Wednesday  January  
3316    3   Thursday  January  
3681    4     Friday  January  
4046    5   Saturday  January  
4412    0     Monday  January  
4777    1    Tuesday  January  
5142    2  Wednesday  January  
5507    3   Thursday  January  
5873    5   Saturday  January  
6238    6     Sunday  January  
6603    0     Monday  January  
6968    1    Tuesday  January  
7334    3   Thursday  January

## A) What was the average fahrenheit for the day? 

In [9]:
birthday = weather_df.query('day == "January 30"') 
temp_avg = (birthday['high'] + birthday['low'])/2
weather_df = weather_df.assign(temp_avg=(weather_df['high']+weather_df['low'])/2)

In [10]:
weather_df.query('day =="January 30"')

day  year  high  low  rain  snow       date  day_num  month_num  \
29    January 30  2005    36   26  0.15   1.3 2005-01-30       30          1   
394   January 30  2006    62   37  0.00   0.0 2006-01-30       30          1   
759   January 30  2007    39   20  0.02   0.2 2007-01-30       30          1   
1124  January 30  2008    49   29  0.05   0.0 2008-01-30       30          1   
1490  January 30  2009    36   22  0.00   0.0 2009-01-30       30          1   
1855  January 30  2010    21   16  0.10   2.1 2010-01-30       30          1   
2220  January 30  2011    39   18  0.00   0.0 2011-01-30       30          1   
2585  January 30  2012    42   33  0.00   0.0 2012-01-30       30          1   
2951  January 30  2013    68   38  0.32   0.0 2013-01-30       30          1   
3316  January 30  2014    28   10  0.00   0.0 2014-01-30       30          1   
3681  January 30  2015    42   20  0.00   0.1 2015-01-30       30          1   
4046  January 30  2016    41   26  0.00   0.0 2016-01-30       30          1   
4412  January 30  2017    39   27  0.02   0.8 2017-01-30       30          1   
4777  January 30  2018    38   22  0.14   0.3 2018-01-30       30          1   
5142  January 30  2019    35    7  0.01   0.2 2019-01-30       30          1   
5507  January 30  2020    39   25  0.00   0.0 2020-01-30       30          1   
5873  January 30  2021    35   19  0.00   0.0 2021-01-30       30          1   
6238  January 30  2022    27   11  0.00   0.0 2022-01-30       30          1   
6603  January 30  2023    54   37  0.00   0.0 2023-01-30       30          1   
6968  January 30  2024    39   34  0.00   0.0 2024-01-30       30          1   
7334  January 30  2025    42   29  0.00   0.0 2025-01-30       30          1   

      dow   dow_name    month  temp_avg  
29      6     Sunday  January      31.0  
394     0     Monday  January      49.5  
759     1    Tuesday  January      29.5  
1124    2  Wednesday  January      39.0  
1490    4     Friday  January      29.0  
1855    5   Saturday  January      18.5  
2220    6     Sunday  January      28.5  
2585    0     Monday  January      37.5  
2951    2  Wednesday  January      53.0  
3316    3   Thursday  January      19.0  
3681    4     Friday  January      31.0  
4046    5   Saturday  January      33.5  
4412    0     Monday  January      33.0  
4777    1    Tuesday  January      30.0  
5142    2  Wednesday  January      21.0  
5507    3   Thursday  January      32.0  
5873    5   Saturday  January      27.0  
6238    6     Sunday  January      19.0  
6603    0     Monday  January      45.5  
6968    1    Tuesday  January      36.5  
7334    3   Thursday  January      35.5

In [12]:
birthday_mean_temp = temp_avg.mean()

In [13]:
birthday_mean_temp

np.float64(32.30952380952381)

#### Since I was born in 2006, combining the weather data of Philadelphia for the past twenty years, the average fahrenheit on my birthday in Philadelphia is 32.3 degrees. 

## B) Did it rain? 

In [14]:
birthday = weather_df.query('day == "January 30"') 
is_rain = birthday['rain']>0

In [15]:
is_rain

29       True
394     False
759      True
1124     True
1490    False
1855     True
2220    False
2585    False
2951     True
3316    False
3681    False
4046    False
4412     True
4777     True
5142     True
5507    False
5873    False
6238    False
6603    False
6968    False
7334    False
Name: rain, dtype: bool

In [16]:
is_rain.value_counts()

rain
False    13
True      8
Name: count, dtype: int64

#### Since I was born in 2006, majority of the years since I was alive on the day of my birthday in Philadelphia it did not rain. (13 years it did not and 7 years it did). 

## C) Did it snow? 

In [19]:
birthday = weather_df.query('day == "January 30"') 
is_snow = birthday['snow']>0

In [20]:
is_snow.value_counts()

snow
False    14
True      7
Name: count, dtype: int64

#### Since I was born in 2006, majority of the years since I was alive on the day of my birthday in Philadelphia it did not snow. (14 years it did not and 6 years it did). 

## D) What day of the week was it? 

In [22]:
weather_df.query('day == "January 30"') ['dow_name'].value_counts()

dow_name
Monday       4
Sunday       3
Tuesday      3
Wednesday    3
Saturday     3
Thursday     3
Friday       2
Name: count, dtype: int64

#### Since I was born in 2006, the largest amount of my birthdays have fallen on a Monday. 

## Conclusion 

The weather in Philadelphia on the days of my birthday for the past twenty years was something I was curious about, especially to see if there was a pattern that I can relate back to myself. I found that majority of my birthdays fell on a Monday, it did snow nor rain for most of the years, and the avgerage temperature was 32 degrees on my birthdays put together. So overall, most of my birthdays throughout the years here in Philadelphia were not extremely cold, but still on the chillier side. I find that interesting, because I spent my recent birthday in 2026 freezing, because of how cold it was from the piled up snow from the previous snow storm and high winds. I thought this was the norm weather here in Philadelphia on Jan 30, but referencing past years' data it shows otherwise. This was an interesting finding.  